Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt

```diff
@@ -0,0 +1,71 @@
+package de.rki.coronawarnapp.ui.interoperability
+
+import androidx.lifecycle.MutableLiveData
+import de.rki.coronawarnapp.storage.interoperability.InteroperabilityRepository
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import testhelpers.extensions.InstantExecutorExtension
+import testhelpers.extensions.getOrAwaitValue
+
+@Extensions(
+    ExtendWith(MockKExtension::class),
+    ExtendWith(InstantExecutorExtension::class)
+)
+class InteroperabilityConfigurationFragmentViewModelTest {
+
+    @MockK(relaxUnitFun = true)
+    lateinit var interoperabilityRepository: InteroperabilityRepository
+
+    private lateinit var interoperabilityConfigurationFragmentViewModel: InteroperabilityConfigurationFragmentViewModel
+
+    private val countryListLiveData = MutableLiveData<List<String>>(listOf())
+
+    @BeforeEach
+    fun setupFreshViewModel() {
+        every { interoperabilityRepository.countryList } returns countryListLiveData
+        interoperabilityConfigurationFragmentViewModel =
+            InteroperabilityConfigurationFragmentViewModel(interoperabilityRepository)
+    }
+
+    @Test
+    fun testNoFetch() {
+        val countryList =
+            interoperabilityConfigurationFragmentViewModel.countryList.getOrAwaitValue()
+
+        assertTrue(countryList.isEmpty())
+    }
+
+    @Test
+    fun testFetch() {
+        val countryListFetched = listOf(
+            "DE", "UK", "FR", "IT", "ES", "PL", "RO", "NL",
+            "BE", "CZ", "SE", "PT", "HU", "AT", "CH", "BG", "DK", "FI", "SK",
+            "NO", "IE", "HR", "SI", "LT", "LV", "EE", "CY", "LU", "MT", "IS"
+        )
+
+        interoperabilityRepository.getAllCountries()
+        countryListLiveData.value = countryListFetched
+
+        val countryList =
+            interoperabilityConfigurationFragmentViewModel.countryList.getOrAwaitValue()
+
+        assertTrue(countryList.isNotEmpty())
+        assertEquals(countryList.size, countryListFetched.size)
```

Comment:
isNotEmpty and size check is checking the same thing

Reply:
never say never :-) but I assert this is true!